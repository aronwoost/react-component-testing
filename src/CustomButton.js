import React from 'react';
import classnames from 'classnames';

class CustomButton extends React.Component {
  render() {
    const props = this.props;
    const classNames = classnames({
      active: props.active,
    });

    return (
      <button className={classNames}>{props.label}</button>
    );
  }
}

export default CustomButton;
