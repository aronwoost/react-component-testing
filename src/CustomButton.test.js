import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import CustomButton from './CustomButton';

it('renders render', () => {
  const component = shallow(
    <CustomButton label='test-label' />
  );

  expect(toJson(component)).toMatchSnapshot();
});

it('renders has class active', () => {
  const component = shallow(
    <CustomButton label='test-label' active={true}/>
  );

  expect(component.hasClass('active')).toBe(true);
});
