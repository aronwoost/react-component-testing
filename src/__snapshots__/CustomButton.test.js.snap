exports[`test renders render 1`] = `
<button
  className="">
  test-label
</button>
`;
